ui/src/app/pages/login/account-creation/account-creation.spec.tsx

```diff
@@ -36,54 +53,96 @@ function findCountryInputField() {
   return screen.getByLabelText('Country input') as HTMLInputElement;
 }
 
+function getAreaOfResearchTextBox(container: HTMLElement) {
+  return container.querySelector('textarea#areaOfResearch');
+}
+
 const defaultConfig = { gsuiteDomain: 'researchallofus.org' };
 
 beforeEach(() => {
   serverConfigStore.set({ config: defaultConfig });
   registerApiClient(ProfileApi, new ProfileApiStub());
-  props = {
-    profile: createEmptyProfile(),
-    onComplete: () => {},
-    onPreviousClick: () => {},
-  };
+});
+
+const fillInMinimalFields = async (user: UserEvent, container: HTMLElement) => {
+  await user.type(screen.getByLabelText(FORM_LABELS.username), 'username');
+  await user.type(screen.getByLabelText(FORM_LABELS.givenName), 'Firstname');
+  await user.type(screen.getByLabelText(FORM_LABELS.familyName), 'Lastname');
+
+  await user.type(
+    screen.getByLabelText(FORM_LABELS.streetAddress1),
+    '1 Main Street'
+  );
+  await user.type(screen.getByLabelText(FORM_LABELS.city), 'Boston');
+  await user.type(findStateField(), 'MA');
+  await user.type(screen.getByLabelText(FORM_LABELS.zipCode), '02115');
+  await user.click(findCountryDropdownField());
+  await user.paste(countryDropdownOption.unitedStates);
+  await user.keyboard('{enter}');
+
+  await user.click(getAreaOfResearchTextBox(container));
+  await user.paste('I am an undergraduate learning genomics.');
+};
+
+it('should allow completing the account creation form', async () => {
+  const onComplete = jest.fn();
+  const { container, user } = setup({
+    ...createProps(),
+    onComplete,
+  });
+
+  await fillInMinimalFields(user, container);
+
+  await user.click(screen.getByLabelText('Next'));
+
+  expect(onComplete).toHaveBeenCalled();
+  const profile: Profile = onComplete.mock.calls[0][0];
+
+  expect(profile.username).toEqual('username');
+  expect(profile.givenName).toEqual('Firstname');
+  expect(profile.familyName).toEqual('Lastname');
+  expect(profile.address.streetAddress1).toEqual('1 Main Street');
+  expect(profile.address.city).toEqual('Boston');
+  expect(profile.address.state).toEqual('MA');
+  expect(profile.address.zipCode).toEqual('02115');
+  expect(profile.address.country).toEqual('United States of America');
+  expect(profile.areaOfResearch).toEqual(
+    'I am an undergraduate learning genomics.'
+  );
 });
 
 it('should handle given name validity', async () => {
-  const user = userEvent.setup();
-  const { container } = component();
+  const { container, user } = setup();
```

Comment:
Changes to all tests in this file other than the newly added ones are strictly refactors.